src/ResourceManager/AzureBatch/Commands.Batch/Pools/GetBatchPoolNodeCountsCommand.cs

```diff
@@ -0,0 +1,53 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.Batch.Models;
+using System.Management.Automation;
+using Constants = Microsoft.Azure.Commands.Batch.Utils.Constants;
+
+namespace Microsoft.Azure.Commands.Batch
+{
+    [Cmdlet(VerbsCommon.Get, Constants.AzureBatchPoolNodeCounts, DefaultParameterSetName = Constants.AzureBatchPoolNodeCounts), OutputType(typeof(PSPoolNodeCounts))]
+    public class GetBatchPoolNodeCountsCommand : BatchObjectModelCmdletBase
+    {
+        private const int defaultMaxCount = 10;
```

Comment:
because the default is 1000 if not specified. The number of 1000 for list* APIs which allow listing objects in pages. in this case we are talking about listing node state count summaries. Setting 1000 as default effectively says no limit in most cases.

Reply:
But that's fine right? What's the issue with using the same default as the API?

Reply:
For example ListPools sets this to: `Constants.DefaultMaxCount` which is 1000